pag-lexer/src/congruence.rs

```diff
@@ -41,20 +44,28 @@ pub fn approximate_congruence_class(tree: &RegexTree) -> Vec<Intervals> {
                 None => vec![x],
             }
         }
-        Concat(r, s) => {
-            if !r.is_nullable() {
-                approximate_congruence_class(r)
-            } else {
-                meet(
-                    &approximate_congruence_class(r),
-                    &approximate_congruence_class(s),
-                )
+        Concat(children) => {
+            match children[1..]
+                .iter()
+                .zip(children.iter().map(|x| x.is_nullable()))
+                .try_fold(
+                    approximate_congruence_class(&children[0]),
+                    |acc, (tree, prev_nullable)| {
+                        if !prev_nullable {
```

Comment:
Why we require `prev_nullable`?

Reply:
![telegram-cloud-photo-size-1-4934244573983648600-x](https://github.com/SchrodingerZhu/paguroidea/assets/20108837/ba7f98a4-139e-4604-a606-730ae8b8e60a)